### ISSUE TITLE
Euclidean metric gives wrong results

### ISSUE THREAD

**Issue by tuxcell** [scikit-learn-contrib/hdbscan#306]
Using the metric 'euclidean' works fine if used with the module implementation. It does not work, in some cases, for precomputed values or for a user-defined euclidean distance function.
Example:

import hdbscan
import numpy as np
import pandas as pd
from sklearn.metrics.pairwise import pairwise_distances

minClusterSize = 2
tmp = [0,0,1,1,3,6,10] # but AOK for [0,0,1,1, 1,3,6,10]
blobs = np.c_[tmp,tmp]
pd.DataFrame(blobs)

#	0	1
#0	0	0
#1	0	0
#2	1	1
#3	1	1
#4	3	3
#5	6	6
#6	10	10

distance = pairwise_distances(blobs, metric='euclidean')
clusterer = hdbscan.HDBSCAN(min_cluster_size=minClusterSize, metric='precomputed')
db = clusterer.fit(distance.astype('float64'))
print(db.labels_)
print(db.probabilities_)
print(db.cluster_persistence_)
print(db.outlier_scores_)

#[-1 -1 -1 -1 -1 -1 -1]
#[0. 0. 0. 0. 0. 0. 0.]
#[]
#[0. 0. 0. 0. 0. 0. 0.]

clusterer2 = hdbscan.HDBSCAN(min_cluster_size=minClusterSize, metric='euclidean')
db2 = clusterer2.fit(blobs)
print(db2.labels_)
print(db2.probabilities_)
print(db2.cluster_persistence_)
print(db2.outlier_scores_)

#[ 0  0  1  1 -1 -1 -1]
#[1. 1. 1. 1. 0. 0. 0.]
#[0. 0.]
#[0.         0.         0.         0.         0.5        0.75    0.85714286]

def mydist(p1, p2):
    diff = p1 - p2
    return np.vdot(diff, diff) ** 0.5
clusterer5 = hdbscan.HDBSCAN(min_cluster_size=minClusterSize, metric=mydist)
db5 = clusterer5.fit(blobs)
print(db5.labels_)
print(db5.probabilities_)
print(db5.cluster_persistence_)
print(db5.outlier_scores_)

#[-1 -1 -1 -1 -1 -1 -1]
#[0. 0. 0. 0. 0. 0. 0.]
#[]
#[0. 0. 0. 0. 0. 0. 0.]
